redash/models/__init__.py

```diff
@@ -821,20 +821,33 @@ def evaluate(self):
     def subscribers(self):
         return User.query.join(AlertSubscription).filter(AlertSubscription.alert == self)
 
-    def render_template(self):
-        if not self.template:
+    def render_template(self, template):
+        if not template:
             return ''
         data = json_loads(self.query_rel.latest_query_data.data)
-        context = {'rows': data['rows'], 'cols': data['columns'], 'state': self.state}
-        return mustache_render(self.template, context)
+        host = base_url(self.query_rel.org);
+        context = {
+            ALERT_NAME: self.name,
+            ALERT_URL: '{host}/alerts/{alert_id}'.format(host=host, alert_id=self.id),
+            ALERT_STATUS: self.state,
+            ALERT_EVALUATION_TIME: datetime.datetime.now(), # could be problematic cuz utc?
+            ALERT_CONDITION: self.options['op'],
+            ALERT_THRESHOLD: self.options['value'],
+            QUERY_NAME: self.query_rel.name,
+            QUERY_URL: '{host}/queries/{query_id}'.format(host=host, query_id=self.query_rel.id),
+            QUERY_RESULT_VALUE: data['rows'][0][self.options['value']],
```

Comment:
1. This is unsafe as `rows` might be empty or it might not have the the specified column. 
2. `self.options['value']` is not the column name.

Reply:
Fixed in https://github.com/getredash/redash/pull/4170/commits/7bbe27f45345398fae7f27165134f75be32e0e59